optin/reporting/test/models/completed.spec.js

```diff
@@ -0,0 +1,204 @@
+const runDBNotFoundTests = require('./runDBNotFoundTests');
+const runTotalRecordTests = require('./runTotalRecordTests');
+const Config = require('../../config');
+const completedDB = require('../../models/completed');
+const { refreshDBs } = require('../../models/database');
+const { expect } = require('chai');
+const { stub, restore } = require('sinon');
+const path = require('path');
+
+const runBasicOptinTypeTests = (types, parameters) => {
+	/* eslint-disable no-await-in-loop */
+	for (let i = 0; i <= types.length - 1; ++i) {
+		// Arrange:
+		const optinType = types[i];
+
+		it(`returns ${0 === optinType ? 'pre' : 'post'}-optin records`, async () => {
+			// Act:
+			const result = await completedDB.getCompletedPagination({
+				...parameters,
+				optinType
+			});
+
+			// Assert:
+			result.forEach(item => {
+				expect(item.isPostoptin).to.be.equal(optinType);
+			});
+		});
+	}
+};
+
+describe('completed models', () => {
+	const parameters = {
+		nemAddressHex: null,
+		optinType: null,
+		pageNumber: 1,
+		pageSize: 10,
+		sortBy: '',
+		sortDirection: '',
+		symbolAddressHex: null,
+		txHash: null
+	};
+
+	beforeEach(() => {
+		stub(Config, 'getDataStoragePath').returns(path.join(__dirname, '../mock'));
+		refreshDBs();
+	});
+
+	afterEach(restore);
+
+	describe('verify if database exists', () => {
+		runDBNotFoundTests({
+			paginationFunc: completedDB.getCompletedPagination,
+			parameters
+		});
+	});
+
+	describe('getCompletedPagination', () => {
+		describe('optin type', () => {
+			runBasicOptinTypeTests([0, 1], parameters);
+		});
+
+		it('returns basic records', async () => {
+			// Arrange + Act:
+			const result = await completedDB.getCompletedPagination({
+				...parameters
+			});
+
+			// Assert:
+			result.forEach(item => {
+				expect(item).to.have.property('id');
+				expect(item).to.have.property('isPostoptin');
+				expect(item).to.have.property('symbolTimestamp');
+				item.nemSource.forEach(source => {
+					expect(source).to.have.property('address');
+					expect(source).to.have.property('balance');
+					expect(source).to.have.property('hashes');
+					expect(source).to.have.property('height');
+					expect(source).to.have.property('timestamps');
+					expect(source).to.have.property('label');
+				});
+				item.symbolDestination.forEach(source => {
+					expect(source).to.have.property('address');
+					expect(source).to.have.property('balance');
+					expect(source).to.have.property('hashes');
+					expect(source).to.have.property('height');
+					expect(source).to.have.property('timestamps');
+				});
+			});
+		});
+
+		it('returns sorted id in descending given invalid sortBy', async () => {
+			// Act:
+			const sort = {
+				sortBy: 'invalid',
+				sortDirection: 'DESC'
+			};
+			const result = await completedDB.getCompletedPagination({
+				...parameters,
+				...sort
+			});
+
+			// Assert:
+			const firstTimestamps = result[0].id;
+			const lastTimestamps = result[result.length - 1].id;
+
```

Comment:
for sorting tests, you should check predicate holds for ALL items in result, not just first and last